cipher-traits/src/algorithms.rs

```diff
@@ -0,0 +1,81 @@
+pub mod keyed_hash_incorrect_hmac_blake2b {
+    use crate::primitives::keyed_hash::*;
+
+    pub const KEY_LEN: usize = 32;
+    pub const OUT_LEN: usize = 32;
+
+    pub trait KeyedHashIncorrectHmacBlake2b: KeyedHash<KEY_LEN, OUT_LEN> {}
+}
+
+pub mod keyed_hash_blake2b {
+    use crate::primitives::keyed_hash::*;
+
+    pub const KEY_LEN: usize = 32;
+    pub const OUT_LEN: usize = 32;
+
+    pub trait KeyedHashBlake2b: KeyedHash<KEY_LEN, OUT_LEN> {}
+}
+
+pub mod keyed_hash_shake256 {
+    use crate::primitives::keyed_hash::*;
+
+    pub const KEY_LEN: usize = 32;
+    pub const OUT_LEN: usize = 32;
+
+    pub trait KeyedHashShake256: KeyedHash<KEY_LEN, OUT_LEN> {}
+}
+
+pub mod aead_chacha20poly1305 {
+    use crate::primitives::aead::*;
+
+    pub const KEY_LEN: usize = 32;
+    pub const NONCE_LEN: usize = 12;
+    pub const TAG_LEN: usize = 16;
+
+    pub trait AeadChaCha20Poly1305: Aead<KEY_LEN, NONCE_LEN, TAG_LEN> {}
+}
+
+pub mod aead_xchacha20poly1305 {
+    use crate::primitives::aead::*;
+
+    pub const KEY_LEN: usize = 32;
```

Comment:
When you have links to the sources for kyber and mceliece, why not for these others?